src/components/DropdownMultipleHierarchy/index.tsx

```diff
@@ -0,0 +1,194 @@
+                props.onChange(newValues)
+            } else {
+                const newValues = deduplicate([
+                    ...props.values,
+                    ...getValueArrWithChildren(item)
+                ])
+                props.onChange(newValues)
+            }
+        },
+        [props.values, props.onChange]
+    )
+
+    // TODO: filter実装
+    // const filteredItems = React.useMemo(() => {
+    //     const items = props.items.filter(item =>
+    //         item.label.includes(searchValue)
+    //     )
+    //     return items
+    // }, [searchValue, props.items])
+
+    const keyDownInInput = React.useCallback(
+        (e: React.KeyboardEvent) => {
+            if (e.keyCode === 8 && searchValue === '') {
+                const slicedItem = props.values.slice(0, -1)
+                props.onChange(slicedItem)
+            }
+        },
+        [props.values, searchValue, props.onChange]
+    )
+
+    const noop = () => {}
+
+    return (
+        <Wrap className={props.className} width={props.width}>
+            <Inner>
+                <ClickOutside.Component
+                    onClickOutside={clickOutside}
+                    inactive={!isMenuVisible}
+                >
+                    <Body.Component
+                        onClick={!props.disabled ? clickBody : noop}
+                        items={props.items}
+                        values={props.values}
+                        placeholder={props.placeholder}
+                        isMenuVisible={isMenuVisible}
+                        diff={props.diff}
+                        disabled={props.disabled}
+                        isError={props.isError}
+                        onChangeSearchValue={changeSearchValue}
+                        searchValue={searchValue}
+                        onKeydown={keyDownInInput}
+                        onClickIcon={!props.disabled ? clickIcon : noop}
+                    />
+                    <StyledMenu
+                        isVisible={isMenuVisible}
+                        items={props.items}
+                        selectedValues={props.values}
+                        onClickItem={changeValue}
+                    />
+                </ClickOutside.Component>
+            </Inner>
+            <ErrorMessage.Component
+                errored={props.isError}
+                message={props.errorMessage}
+            />
+        </Wrap>
+    )
+})
+
+const deduplicate = (arr: string[]) => Array.from(new Set(arr))
+
+const getValueArrWithChildren = (item: Menu.Item) => {
+    let resultArr: Menu.Value[] = []
+    resultArr.push(item.value)
+
+    if (item.children === undefined) return resultArr
+
+    for (const child of item.children) {
+        if (child.children === undefined) {
+            resultArr.push(child.value)
+        } else {
+            resultArr = resultArr.concat(getValueArrWithChildren(child))
+        }
```

Comment:
IMO: else文個人的に好きじゃないので、

```suggestion
        if (child.children === undefined) {
            resultArr.push(child.value)
            continue;
        }
        
        resultArr = resultArr.concat(getValueArrWithChildren(child))
```
とかどうですか？

Reply:
好きか嫌いかよりも、elseを使わないで欲しい理由や使うことで懸念されることを言語化できると受け手に納得されやすくなると思います 👍

Reply:
if..elseだと、ifの条件に引きずられたelse文の中を考えないといけなくなり、if文の条件を考慮してしまうことになります。
早期リターンのように早めにifで切り抜けられると、後続の記述はif文の条件を考慮せずに済むので、可読性の向上に繋がります。

Reply:
57f17ab 修正しました 🙆‍♀️